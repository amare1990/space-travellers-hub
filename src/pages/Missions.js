import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { reserveMission, fetchMissions, cancelMission } from '../redux/missions/missions';
import '../styles/Missions.css';

const Missions = () => {
  const missionsArray = useSelector((state) => state.missions.missions);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!missionsArray.length) {
      dispatch(fetchMissions());
    }
  });

  return (

    <div className="missions-wrapper">

      <li className="mission-card">
        <div className="missionName"><h3>Mission</h3></div>
        <div className="missionDescription"><h3>Description</h3></div>
        <div className="missionMember"><h3>Status</h3></div>
        <div className="missionButton" />
      </li>

      { missionsArray.map((mission) => (

        <li key={mission.mission_id} className="mission-card">
          <div className="missionName"><h3>{mission.mission_name}</h3></div>
          <div className="missionDescription"><p>{mission.description}</p></div>
          <div className="missionMember">
            {mission.reserved ? (<button type="button" className="missionMemberActive">Active MEMBER</button>)
              : (<button type="button" className="missionMemberNot">NOT A MEMBER</button>)}
          </div>
          <div className="missionButton">
            {mission.reserved ? (
              <button
                type="button"
                className="leavebutton"
                onClick={() => dispatch(cancelMission(mission.mission_id))}
              >
                Leave Mission
              </button>
            )
              : (
                <button
                  type="button"
                  onClick={() => dispatch(reserveMission(mission.mission_id))}
                >
                  Join Mission
                </button>
              )}
          </div>
        </li>
      ))}

    </div>
  );
};

export default Missions;
