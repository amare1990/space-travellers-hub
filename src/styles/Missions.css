.missions-wrapper {
  margin: 20px 120px 0 200px;
}

.mission-card {
  list-style-type: none;
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin: auto;
}

.mission-card:nth-child(1n) {
  background-color: white;
}

.mission-card:nth-child(2n) {
  background-color: lightgray;
}

.missionName {
  width: 10vw;
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  border: solid 1px #444;
}

.missionName h3 {
  text-align: left;
  margin-left: 5px;
}

.missionDescription {
  width: 60vw;
  border: solid 1px #444;
}

.missionDescription p {
  margin: 5px;
}

.missionMember {
  width: 10vw;
  display: flex;
  justify-content: center;
  align-items: center;
  border: solid 1px #444;
}

.missionMemberNot {
  background-color: #666;
  color: white;
  border: none;
  border-radius: 5px;
}

.missionMemberActive {
  background-color: rgb(3, 201, 164);
  color: white;
  border: none;
  border-radius: 5px;
}

.missionButton {
  width: 10vw;
  display: flex;
  justify-content: center;
  align-items: center;
  border: solid 1px #444;
}

.missionButton button {
  background-color: rgba(255, 255, 255, 0);
  border-radius: 5px;
  height: 40px;
}

.leavebutton {
  color: red;
  border: red solid;
}
