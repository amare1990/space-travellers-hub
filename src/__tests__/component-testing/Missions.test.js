import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Missions from '../../pages/Missions';

test('Testing if Missions component renders correctly', () => {
  const missionsSnapshot = renderer.create(
    <Provider store={store}>
      <Missions />
    </Provider>,
  ).toJSON();
  expect(missionsSnapshot).toMatchSnapshot();
});
