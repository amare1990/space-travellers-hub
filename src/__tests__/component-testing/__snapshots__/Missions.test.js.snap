// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing if Missions component renders correctly 1`] = `
<div
  className="missions-wrapper"
>
  <li
    className="mission-card"
  >
    <div
      className="missionName"
    >
      <h3>
        Mission
      </h3>
    </div>
    <div
      className="missionDescription"
    >
      <h3>
        Description
      </h3>
    </div>
    <div
      className="missionMember"
    >
      <h3>
        Status
      </h3>
    </div>
    <div
      className="missionButton"
    />
  </li>
</div>
`;
